### PR TITLE
feat(efb): mcdu keyboard input settings

### LINKED ISSUE #5727: EFB: Move the MCDU Keyboard Input Option to the EFB
### Aircraft Version

Development

### Description

Move the keyboard input option from the MCDU into the EFB settings. It is one of the last 3 remaining "sim" options to be removed from the MCDU after #5344.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### PR DESCRIPTION
**Pull request by 2hwk**
Fixes #5727

## Summary of Changes

* Moved the setting for MCDU keyboard input to the EFB
  * Allows keyboard input to be toggled on for the MCDU
  * Customisable timeout duration (5-120seconds) before inactivity returns the focus to the camera from the MCDU. 
* Removed the requirement for a full reload of the plane to use this - Trade-off is that the MCDU screen is now always considered a touchscreen input by the sim even when the setting is disabled.

Note: May need a change to the relevant docs for this feature as well

## Screenshots (if necessary)

image

image

## Additional context

Discord username (if different from GitHub): 2Cas#1022

## Testing instructions

- Spawn in, turn on EFB, Go to EFB Settings > Sim Options
- Enable keyboard input
- Click anywhere on the MCDU screen, the header should highlight cyan, the camera controls will now be locked and the keyboard input will be captured by the MCDU scratchpad.
- Click anywhere on the MCDU to undo or press alt or ctrl+z or wait for the timeout (default 60s) to finish.
- Disable keyboard input
- MCDU should no longer accept input when clicked
- Check that this setting persists during and after flights. This setting should persist indefinitely once set. 

Known Issues:
- ~~Setting a value outside of the 5-120 second range will appear to accept on the input field. This is not saved, when you switch from one EFB setting to another, this will reset back to the last accepted time (between 5 and 120 seconds)~~

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by 2hwk** [comment]
This PR requires #5741 to be merged first.

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
         return "A320_Neo_CDU";
     }
     get isInteractive() {
-        return NXDataStore.get("MCDU_KB_INPUT", "DISABLED") === "ENABLED";
+        return true;
     }
```
Comment (Benjozork) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:93]: **question:** can you explain why this is needed ?

File: src/instruments/src/EFB/Settings/Settings.tsx [hunk 2]
```diff
@@ -186,6 +189,49 @@ const SimOptionsPage = () => {
...
+        <div className="py-4 flex flex-row justify-between items-center">
+            <span>
+                <span className="text-lg text-gray-300">MCDU Keyboard Input</span>
+                <span className="text-lg text-gray-500 ml-2">(unrealistic)</span>
+            </span>
+            <Toggle value={mcduInput === 'ENABLED'} onToggle={(value) => setMcduInput(value ? 'ENABLED' : 'DISABLED')} />
+        </div>
+    );
+
+    const mcduTimeoutSetting = (mcduInput === 'ENABLED')
```
Comment (Benjozork) [inline src/instruments/src/EFB/Settings/Settings.tsx:202]: **suggestion:** I would extract those to separate components

**Review by Benjozork — CHANGES_REQUESTED** [review]

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js (hunk 1 above)
Comment (2hwk) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:93]: called by the sim to determine if the display is a touchscreen or not, just return true always as this is more simple and doesn't require a reload to work.

**Comment by tracernz** [comment]
Does it need a changelog?

**Comment by 2hwk** [comment]
> Does it need a changelog?

Probably not, not a big enough change to justify it

File: src/instruments/src/EFB/Settings/Settings.tsx (hunk 2 above)
Comment (2hwk) [reply src/instruments/src/EFB/Settings/Settings.tsx]: solved by modifying SimpleInput to have proper logic for disabling <input>, a better solution than making another component.

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js [hunk 3]
```diff
@@ -920,13 +920,13 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
     initKeyboardScratchpad() {
         window.document.addEventListener('click', () => {
 
-            const mcduInput = NXDataStore.get("MCDU_KB_INPUT", "DISABLED");
+            const mcduInput = NXDataStore.get("CONFIG_MCDU_KB_INPUT", "0");
```
Comment (Benjozork) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:923]: Aren't we now breaking compatibility with the old setting?

**Review by Benjozork — APPROVED** [review]

**Comment by FlightSimmer7700** [comment]
Quality Assurance Tester

Discord: FlightSimmer7700#3468
Object of testing: #5754
Tier of Testing: 1
Date: 5/9/21

Testing Process:
- Loaded in C&D
- tested typing
- changed settings in efb to make sure they applied immediately
- tested times and clicking on and off
- checked that setting was removed from mcdu options
- completed the full flight
- closed the sim
- reopened the sim and loaded in to check persistance in "timer" for keyboard input

Negatives: N/A
Testing Results: Passed

Conclusions: LGTM
Media:
image

**Comment by NUTTTTTTTT** [comment]
Quality Assurance Tester Report

Discord: █▀█ █▄█ ▀█▀#2123
Object of testing: #PR 5754
Tier of Testing: 1
Date: 05/09/2021

Testing Process:
- Loaded in on the runway at EGCC.
- Checked keyboard input active as before, verified no change to previous setting.
- Opened EFB > settings > Sim Options > Disabled MCDU keyboard input and failed to enter input mode on MCDU - PASS
- Re-enabled MCDU keyboard input, checked enabled again - PASS
- Changed MCDU focus timeout and verified times out and closes keyboard input - PASS.
- Disabled once more, reloaded sim and loaded back in C+D at gate, MCDU input and timeout setting persistence checked - PASS

Negatives: N/A
Testing Results: PASS 

Conclusions: Works as described following testing instructions, roughly checked over other EFB functions, all seem to be working okay, no other issues noted - Great stuff!

Media: 

image

**Pull request merged**